### PR TITLE
copr: use fat bytes array

### PR DESCRIPTION
**Pull request by BusyJay**
@ngaut @coocood @shenli @qiuyesuifeng PTAL

### CODE REVIEW

File: tests/coprocessor/test_select.rs [hunk 1]
```diff
@@ -28,6 +27,64 @@ fn next_id() -> i64 {
...
+        }
+    }
+}
+
+impl Iterator for ChunkSpliter {
+    type Item = Row;
+
+    fn next(&mut self) -> Option<Row> {
+        loop {
+            match self.chunk.first() {
```
Comment (ngaut) [inline tests/coprocessor/test_select.rs]: Using "if return" pattern.

File: src/server/coprocessor/endpoint.rs [hunk 2]
```diff
@@ -583,58 +599,47 @@ impl<'a> SelectContext<'a> {
...
-            if rows.len() >= limit {
+            if collected >= limit {
                 break;
             }
             let timer = Instant::now();
-            let ran_rows = try!(self.get_rows_from_range(ran, limit - rows.len(), desc));
-            debug!("fetch {} rows takes {} ms",
-                   ran_rows.len(),
+            let handle_cnt = try!(self.get_rows_from_range(ran, limit, desc));
+            debug!("fetch {} chunks takes {} ms",
```
Comment (coocood) [inline src/server/coprocessor/endpoint.rs]: rename `handle_cnt` to `row_cnt`, `chunks` to `rows`?

**Comment by coocood** [comment]
LGTM

**Comment by shenli** [comment]
@tiancaiamao PTAL

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**